### PR TITLE
Fix streaming auth locally

### PR DESCRIPTION
**Pull request by Weves**
Allows local dev with auth enabled + using streaming endpoints

### CODE REVIEW

File: web/src/app/api/[...path]/route.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+      { status: 404 }
+    );
+  }
+
+  try {
+    const backendUrl = new URL(`http://localhost:8080/${path.join("/")}`);
+    const requestBody = request.body
+      ? JSON.stringify(await request.json())
+      : null;
+    // console.log(requestBody)
```
Comment (pablonyx) [inline web/src/app/api/[...path]/route.ts]: Nit: should likely remove this

File: web/src/app/api/[...path]/route.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      {
+        message:
+          "This API is only available in development mode. In production, something else (e.g. nginx) should handle this.",
+      },
+      { status: 404 }
+    );
+  }
+
+  try {
+    const backendUrl = new URL(`http://localhost:8080/${path.join("/")}`);
```
Comment (pablonyx) [inline web/src/app/api/[...path]/route.ts]: Nit: likely worth importing and using `INTERNAL_URL` in case API server is running on a different port

**Review by pablodanswer — CHANGES_REQUESTED** [review]

File: web/src/app/api/[...path]/route.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    const requestBody = request.body
+      ? JSON.stringify(await request.json())
```
Comment (pablonyx) [inline web/src/app/api/[...path]/route.ts]: Believe we need to add additional handling for different content types (specifically I know we're using formData, url params in a couple places)

File: web/src/app/api/[...path]/route.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+      const { readable, writable } = new TransformStream();
+      response.body?.pipeTo(writable);
+
+      return new NextResponse(readable, {
+        status: response.status,
+        headers: response.headers,
+      });
+    } else {
+      // If it's not a stream, handle it as before
+      const data = await response.text();
```
Comment (pablonyx) [inline web/src/app/api/[...path]/route.ts]: Don't think it's valid to assume we always want .text() - (e.g. for `204` or any other content types)

File: web/src/app/api/[...path]/route.ts (hunk 2 above)
Comment (Weves) [reply web/src/app/api/[...path]/route.ts]: Good suggestion—added!

File: web/src/app/api/[...path]/route.ts (hunk 3 above)
Comment (Weves) [reply web/src/app/api/[...path]/route.ts]: Added handling for formData! url params should be handled above 👀

File: web/src/app/api/[...path]/route.ts (hunk 1 above)
Comment (Weves) [reply web/src/app/api/[...path]/route.ts]: good catch

File: web/src/app/api/[...path]/route.ts (hunk 4 above)
Comment (Weves) [reply web/src/app/api/[...path]/route.ts]: Added handling for images. I believe 204 should be handled properly (.text() will just be empty string)

File: web/src/app/api/[...path]/route.ts [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        });
+      } else {
+        // For other types, handle as before
+        const data = await response.text();
+
+        return new NextResponse(data, {
+          status: response.status,
+          headers: {
+            "Content-Type": contentType,
+          },
```
Comment (pablonyx) [inline web/src/app/api/[...path]/route.ts]: Think we're not handling all the data types (e.g. 204 response code, json content type) - maybe we could have a stricter check for using response.text() and default to response.body more often?

Comment (Weves) [reply web/src/app/api/[...path]/route.ts]: I believe those should all be handled as expected (e.g. all our APIs return JSONs, but it all works as expected + 204 is "no content" which is handled fine here as well). Afaik most response types are just text / can be represented fully as just text

Comment (Weves) [reply web/src/app/api/[...path]/route.ts]: using body by default now

**Pull request merged**